msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -346,10 +346,12 @@ private static Authority createDefaultAADAuthority() {
         readTimeoutForDefaultHttpClient = builder.readTimeoutForDefaultHttpClient;
         serviceBundle = new ServiceBundle(
                 builder.executorService,
+                new TelemetryManager(telemetryConsumer, builder.onlySendFailureTelemetry),
                 builder.httpClient == null ?
                         new DefaultHttpClient(builder.proxy, builder.sslSocketFactory, builder.connectTimeoutForDefaultHttpClient, builder.readTimeoutForDefaultHttpClient) :
                         builder.httpClient,
-                new TelemetryManager(telemetryConsumer, builder.onlySendFailureTelemetry));
+                this instanceof ManagedIdentityApplication ? new HttpHelperManagedIdentity() : new HttpHelper()
```

Comment:
It's generally a red flag for a base class to be aware of its implementations. https://softwareengineering.stackexchange.com/questions/219543/should-a-class-know-about-its-subclasses

Is it possible to composition for this instead?

Reply:
Agreed, I just did it for simplicity and to avoid (nearly) duplicate code, but it's definitely a bad practice. 

In the latest commit I did a proper refactor, and moved the ServiceBundle creation down into the implementation classes so these sorts of 'instanceof' checks aren't needed.